src/ubase/trace.ml

```diff
@@ -121,7 +125,16 @@ let logfile =
 
 let logch = ref None
 
+let closelog _ =
+  match !logch with
+    None -> ()
+  | Some(ch,file) ->
+      close_out ch;
+      logch := None;
+  ;;
+
```

Comment:
Just to illustrate what I had in mind.
```suggestion
      logch := None

let sigusr1_behavior = signal sigusr1 (Signal_handle closelog)
```
And then of course remove the `Signal_ignore` and installing the handler in function `getLogch`.

Reply:
Before reading this I was already busy patching and testing, please check if my latest tweaks match your expectations.

Reply:
No worries. Looks good to me.